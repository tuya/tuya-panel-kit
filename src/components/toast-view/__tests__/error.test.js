import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Toast from '../index';

describe('Toast.Error Component', () => {
  it('basic render', () => {
    const component = renderer
      .create(<Toast.Error show={true} text="I'm Iron man" onFinish={jest.fn()} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
