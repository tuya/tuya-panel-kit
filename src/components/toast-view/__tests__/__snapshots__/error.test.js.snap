// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast.Error Component basic render 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      false,
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      null,
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,.7)",
        "borderRadius": 24,
        "justifyContent": "center",
        "marginBottom": 64,
        "marginHorizontal": 132,
        "opacity": 0,
        "paddingHorizontal": 36,
        "paddingVertical": 36,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 84,
            "width": 84,
          },
          false,
          null,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 84,
              "width": 84,
            },
          ]
        }
        vbHeight={1024}
        vbWidth={1024}
      >
        <RNSVGPath
          d="M512 0c282.7776 0 512 229.2224 512 512s-229.2224 512-512 512S0 794.7776 0 512 229.2224 0 512 0z m0 51.2C257.5104 51.2 51.2 257.5104 51.2 512s206.3104 460.8 460.8 460.8 460.8-206.3104 460.8-460.8S766.4896 51.2 512 51.2z m162.9184 297.8816a38.4 38.4 0 0 1 0 54.2976L566.3232 512l108.5952 108.6464a38.4 38.4 0 1 1-54.2976 54.2976L512 566.2976l-108.6464 108.6208a38.4 38.4 0 1 1-54.2976-54.2976l108.6208-108.6464-108.6208-108.5952a38.4 38.4 0 1 1 54.2976-54.2976l108.6464 108.5952 108.5952-108.5952a38.4 38.4 0 0 1 54.2976 0z"
          fill={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "fill",
            ]
          }
          stroke={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
        />
      </RNSVGSvgView>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "flexWrap": "wrap",
            "fontSize": 14,
            "lineHeight": 20,
            "textAlign": "center",
          },
          Object {
            "marginTop": 12,
          },
          null,
        ]
      }
    >
      I'm Iron man
    </Text>
  </View>
</View>
`;
